kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +574,67 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Like like) {
+            PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(like);
+            ColumnVector left = argResults.leftResult;
+            ColumnVector right = argResults.rightResult;
+            int numRows = argResults.rowCount;
+            boolean[] result = new boolean[numRows];
+            boolean[] nullability = new boolean[numRows];
+            for (int rowId = 0; rowId < numRows; rowId++) {
+                String leftOp = left.getString(rowId);
+                String rightOp = right.getString(rowId);
+                if (!Objects.isNull(leftOp) && !Objects.isNull(rightOp)) {
+                    nullability[rowId] = false;
+                    String rightOpRegexPattern =
+                            escapeLikeRegex(rightOp, like.getEscape());
+                    result[rowId] = leftOp.matches(rightOpRegexPattern);
+                } else {
+                    nullability[rowId] = true;
+                    result[rowId] = false;
+                }
+            }
+            return new DefaultBooleanVector(numRows,
+                    Optional.of(nullability), result);
+        }
+
+        private String escapeLikeRegex(String pattern, char escape) {
```

Comment:
Create a `LikeEvaluator` similar to `ElementAtEvaluator` and move all this logic there.

Reply:
done.